tests/terminator_tests/improvement_tests/test_preprocessing.py

```diff
@@ -0,0 +1,236 @@
+    # Scenario: categorical parameters are mapped to the float distribution.
+    trials_before = [
+        create_trial(
+            params={"categorical": 0},
+            distributions={"categorical": CategoricalDistribution((0, 1))},
+            value=1.0,
+        ),
+        create_trial(
+            params={"categorical": 1},
+            distributions={"categorical": CategoricalDistribution((0, 1))},
+            value=1.0,
+        ),
+    ]
+    p = _preprocessing.OneToHot()
+    trials_after = p.apply(trials_before, direction)
+
+    for t in trials_after:
+        # Distributions are expected to be mapped to `FloatDistribution(0, 1)` for each choice.
+        assert len(t.distributions) == 2
+        for d in t.distributions.values():
+            assert d == FloatDistribution(0, 1)
+
+    assert trials_after[0].params == {"i0_categorical": 1.0, "i1_categorical": 0.0}
+    assert trials_after[1].params == {"i0_categorical": 0.0, "i1_categorical": 1.0}
+
+    # Scenario: int and float parameters are identical even after the preprocessing.
+    trials_before = [
+        create_trial(
+            params={
+                "float": 1,
+                "int": 3,
+            },
+            distributions={
+                "float": FloatDistribution(1, 10),
+                "int": IntDistribution(1, 10),
+            },
+            value=1.0,
+        ),
+    ]
+    p = _preprocessing.OneToHot()
+    trials_after = p.apply(trials_before, direction)
+
+    for name_before, param_before in trials_before[0].params.items():
+        distribution_before = trials_before[0].distributions[name_before]
+
+        # Note that the param/distribution names are modified with "i0_" prefix to be consistent
+        # with categorical parameters.
+        name_after = f"i0_{name_before}"
+        param_after = trials_after[0].params[name_after]
+        distribution_after = trials_after[0].distributions[name_after]
+
+        assert distribution_before == distribution_after
+        assert param_before == param_after
+
+
+@pytest.mark.parametrize("direction", (StudyDirection.MINIMIZE, StudyDirection.MAXIMIZE))
+def test_add_random_inputs(direction: StudyDirection) -> None:
+    n_additional_trials = 3
+    dummy_value = -1
+    distributions = {
+        "bacon": CategoricalDistribution((0, 1, 2)),
+        "egg": FloatDistribution(1, 10),
+        "spam": IntDistribution(1, 10),
+    }
+
+    trials_before = [
+        create_trial(
+            params={"bacon": 0, "egg": 1, "spam": 10},
+            distributions=distributions,
+            value=1.0,
+        ),
+    ]
+    p = _preprocessing.AddRandomInputs(
+        n_additional_trials=n_additional_trials,
+        dummy_value=dummy_value,
+    )
+    trials_after = p.apply(trials_before, direction)
+
+    assert trials_before[0] == trials_after[0]
+    assert len(trials_after) == len(trials_before) + n_additional_trials
+    for t in trials_after[1:]:
```

Comment:
A nit, but swapping L228 and L229 makes it easier to understand the intent of the test for me.

```suggestion
    assert len(trials_after) == len(trials_before) + n_additional_trials
    assert trials_before[0] == trials_after[0]
    for t in trials_after[1:]:
```